tests/unittests/test_datasource/test_hetzner.py

```diff
@@ -115,3 +119,26 @@ def test_not_on_hetzner_returns_false(self, m_on_hetzner, m_find_fallback,
         # These are a white box attempt to ensure it did not search.
         m_find_fallback.assert_not_called()
         m_read_md.assert_not_called()
+
+
+class TestMaybeB64Decode(CiTestCase):
+    """Test the maybe_b64decode helper function."""
+
+    def test_raises_error_on_non_bytes(self):
+        """maybe_b64decode should raise error if data is not bytes."""
+        with pytest.raises(TypeError):
+            hc_helper.maybe_b64decode(str("this is not bytes"))
+
+        with pytest.raises(TypeError):
+            hc_helper.maybe_b64decode(int(4))
+
+    def test_returns_data_if_not_b64_encoded(self):
+        """If data is not b64 encoded, then return value should be the same."""
+        data = b"regular old data"
+        self.assertEqual(data, hc_helper.maybe_b64decode(data))
+
+    def test_returns_decoded_data_if_encoded(self):
+        """If data is b64 encoded, then return value should be decoded."""
+        data = b"regular old data"
+        self.assertEqual(
+            data, hc_helper.maybe_b64decode(base64.b64encode(data)))
```

Comment:
We probably shouldn't mix `pytest.raises` and `CiTestCase`; even though it works, it's a little confusing. If you wanted to rewrite this as a pytest test class, I think it would look something like this:

```suggestion
class TestMaybeB64Decode:
    """Test the maybe_b64decode helper function."""

    @pytest.mark.parametrize("invalid_input", (str("not bytes"), int(4)))
    def test_raises_error_on_non_bytes(self, invalid_input):
        """maybe_b64decode should raise error if data is not bytes."""
        with pytest.raises(TypeError):
            hc_helper.maybe_b64decode(invalid_input)

    @pytest.mark.parametrize("in_data,expected", [
        # If data is not b64 encoded, then return value should be the same.
        (b"data", b"data"),
        # If data is b64 encoded, then return value should be decoded.
        (base64.b64encode(b"data", b"data")),
    ])
    def test_happy_path(self, in_data, expected):
        assert expected == hc_helper.maybe_b64decode(in_data)
```

(Obviously you could also just `s/pytest.raises/self.assertRaises/`, but where's the fun in that?)

Reply:
adjusted to use pytest only.